File: tree-hugger/lib/tests/fixtures/sample.py
"""Sample Python module for testing symbol extraction."""

import os
from typing import Optional


def greet(name: str = "World") -> str:
    """Greet a person by name.

    Args:
        name: The name of the person to greet. Defaults to "World".

    Returns:
        A formatted greeting string.
    """
    return f"Hello, {name}!"


def greet_many(*names: str) -> None:
    """Greet multiple people.

    Args:
        *names: Variable number of names to greet.
    """
    for name in names:
        print(greet(name))


class Greeter:
    """A class that can generate greetings."""

    def __init__(self, prefix: str = "Hello") -> None:
        """Initialize the Greeter.

        Args:
            prefix: The prefix to use for greetings.
        """
        self.prefix = prefix

    def greet(self, name: str) -> str:
        """Greet a person using this greeter's prefix.

        Args:
            name: The name of the person to greet.

        Returns:
            A formatted greeting string.
        """
        return f"{self.prefix}, {name}!"

File: tree-hugger/lib/tests/fixtures/sample.ts
import { readFile } from "fs";

/**
 * Greets a person by name.
 * @param name - The name of the person to greet
 * @returns A formatted greeting string
 */
export function greet(name: string = "World"): string {
    console.log(readFile);
    return `Hello, ${name}!`;
}

/**
 * Greets multiple people.
 * @param names - Variable number of names to greet
 */
export function greetMany(...names: string[]): void {
    names.forEach(name => console.log(greet(name)));
}

/**
 * A class that can generate greetings.
 */
export class Greeter {
    /** The prefix to use for greetings. */
    private prefix: string;

    /**
     * Creates a new Greeter.
     * @param prefix - The prefix to use for greetings
     */
    constructor(prefix: string = "Hello") {
        this.prefix = prefix;
    }

    /**
     * Greets a person using this greeter's prefix.
     * @param name - The name of the person to greet
     * @returns A formatted greeting string
     */
    greet(name: string): string {
        return `${this.prefix}, ${name}!`;
    }
}

/**
 * Interface for a greeting service.
 */
export interface GreetingService {
    /** Greets a person by name. */
    greet(name: string): string;
}

/** Type alias for a greeting function. */
export type GreetFn = (name: string) => string;

/** Status codes for responses. */
export enum Status {
    Success,
    Error,
    Pending,
}
